topo/topo.go

```diff
@@ -187,3 +194,37 @@ func GetCollStats(ctx context.Context, m *mongo.Client, db, coll string) (*CollS
 
 	return stats, nil
 }
+
+// RunWithRetry executes the provided function with retry logic for transient errors.
+// It retries the function up to maxRetries times,
+// with an exponential backoff starting from retryInterval.
+func RunWithRetry(
+	ctx context.Context,
+	fn func(context.Context) error,
+	retryInterval time.Duration,
+	maxRetries int,
+) error {
+	if retryInterval <= 0 || maxRetries <= 0 {
+		return errors.New("retryInterval and maxRetries must be greater than zero")
+	}
+
+	var err error
+
+	currentInterval := retryInterval
+	for range maxRetries {
+		err = fn(ctx)
+		if err == nil {
+			return nil
+		}
+
+		if !IsTransientError(err) {
+			return err //nolint:wrapcheck
+		}
+
+		log.Ctx(ctx).Warnf("Transient write error: %v, retrying in %s", err, retryInterval)
```

Comment:
It'll not be clear from this log entry which db command had a transient error. E.g. it'll log error:
```
Transient write error: some err, retrying in 2 sec
```
and then it'll pass in the second try, but there will not be trace about original command. When a few commands are executing in parallel it can be misleading. 

Also I would keep `retryInterval`, it can be handy to know it. But agree that currentInterval is mandatory.

Reply:
Hey, totally agree with this one and I have a task to make this a more general solution that will cover other parts of PLM, not just the catalog and there the solution will be a bit more elaborate. For now just wanted to have this working.